src/ngMessages/messages.js

```diff
@@ -320,8 +320,13 @@ angular.module('ngMessages', [])
      var INACTIVE_CLASS = 'ng-inactive';
 
      return {
-       require: 'ngMessages',
        restrict: 'AE',
+       bindToController: {
+         ngMessages: '@',
+         forAttr: '@for'
+         },
+       scope: true,
+       controllerAs: 'ngMessagesController',
```

Comment:
This and `scope: true` are changing the semantics of the directive. Not sure if we want this.